### ISSUE TITLE
Python loads DLLs from incorrect environment

### ISSUE THREAD

**Issue by dihm** [conda/conda#8789]
## Current Behavior
When python is run from a non-base environment, python loads some DLLs from other environments. 

I found this while trying to troubleshoot a related issue described [here](https://github.com/jupyter/notebook/issues/4569), where numpy won't import when run from jupyter but will from an anaconda prompt. Working on that I found that the jupyter kernel was trying to load DLLs from the base environment. 

Researching that, I found this was a more common problem than just with numpy and ipykernels. See the minimal working example below.

### Steps to Reproduce
From an anaconda prompt run
```
conda create -n test python psutil
conda create -n test2 python psutil
conda activate test
python
>>>import os, psutil
>>>p = psutil.Process(os.getpid())
>>>for dll in p.memory_maps():
>>>    print(dll.path)
```
For my collection of environments, I get the following
```
C:\Windows\System32\locale.nls
C:\Windows\System32\en-US\setupapi.dll.mui
C:\Windows\Globalization\Sorting\SortDefault.nls
C:\Windows\System32\en-US\KernelBase.dll.mui
C:\Windows\System32\en-US\kernel32.dll.mui
C:\Windows\System32\user32.dll
C:\Windows\System32\kernel32.dll
C:\Windows\System32\ntdll.dll
C:\Windows\System32\psapi.dll
C:\Miniconda3\envs\pylon\python.exe
C:\Miniconda3\envs\pylon\python37.dll
C:\Miniconda3\envs\pylon\ucrtbase.dll
C:\Windows\System32\pdh.dll
C:\Miniconda3\envs\test\Lib\site-packages\psutil\_psutil_windows.cp37-win_amd64.pyd
C:\Miniconda3\envs\pylon\DLLs\_lzma.pyd
C:\Miniconda3\envs\pylon\DLLs\_bz2.pyd
C:\Miniconda3\envs\pylon\DLLs\select.pyd
C:\Miniconda3\envs\pylon\DLLs\_socket.pyd
C:\Miniconda3\envs\pylon\python3.dll
C:\Miniconda3\envs\pylon\api-ms-win-crt-filesystem-l1-1-0.dll
C:\Miniconda3\envs\pylon\api-ms-win-crt-conio-l1-1-0.dll
C:\Miniconda3\envs\pylon\api-ms-win-crt-process-l1-1-0.dll
C:\Miniconda3\envs\pylon\api-ms-win-crt-environment-l1-1-0.dll
C:\Miniconda3\envs\pylon\api-ms-win-crt-time-l1-1-0.dll
C:\Miniconda3\envs\pylon\api-ms-win-crt-locale-l1-1-0.dll
C:\Miniconda3\envs\pylon\api-ms-win-crt-math-l1-1-0.dll
C:\Miniconda3\envs\pylon\api-ms-win-crt-convert-l1-1-0.dll
C:\Miniconda3\envs\pylon\api-ms-win-crt-stdio-l1-1-0.dll
C:\Miniconda3\envs\pylon\api-ms-win-crt-heap-l1-1-0.dll
C:\Miniconda3\envs\pylon\api-ms-win-crt-string-l1-1-0.dll
C:\Miniconda3\envs\pylon\api-ms-win-core-synch-l1-2-0.dll
C:\Miniconda3\envs\pylon\api-ms-win-core-file-l2-1-0.dll
C:\Miniconda3\envs\pylon\api-ms-win-core-timezone-l1-1-0.dll
C:\Miniconda3\envs\pylon\api-ms-win-core-processthreads-l1-1-1.dll
C:\Miniconda3\envs\pylon\vcruntime140.dll
C:\Miniconda3\envs\pylon\api-ms-win-core-file-l1-2-0.dll
C:\Miniconda3\envs\pylon\api-ms-win-core-localization-l1-2-0.dll
C:\Miniconda3\envs\pylon\api-ms-win-crt-runtime-l1-1-0.dll
C:\Windows\System32\winnsi.dll
C:\Windows\System32\IPHLPAPI.DLL
C:\Windows\System32\powrprof.dll
C:\Windows\System32\version.dll
C:\Windows\System32\wtsapi32.dll
C:\Windows\System32\winsta.dll
C:\Windows\System32\rsaenh.dll
C:\Windows\System32\cryptsp.dll
C:\Windows\System32\cryptbase.dll
C:\Windows\System32\devobj.dll
C:\Windows\System32\cfgmgr32.dll
C:\Windows\System32\KernelBase.dll
C:\Windows\System32\msctf.dll
C:\Windows\System32\nsi.dll
C:\Windows\System32\usp10.dll
C:\Windows\System32\rpcrt4.dll
C:\Windows\System32\shlwapi.dll
C:\Windows\System32\advapi32.dll
C:\Windows\System32\shell32.dll
C:\Windows\System32\oleaut32.dll
C:\Windows\System32\imm32.dll
C:\Windows\System32\msvcrt.dll
C:\Windows\System32\ws2_32.dll
C:\Windows\System32\setupapi.dll
C:\Windows\System32\lpk.dll
C:\Windows\System32\sechost.dll
C:\Windows\System32\gdi32.dll
C:\Windows\System32\ole32.dll
C:\Windows\System32\apisetschema.dll
```
The pylon environment is an older one that I haven't activated in some time.

`sys.path` returns
```
C:\Miniconda3\envs\test\python37.zip
C:\Miniconda3\envs\test\DLLs
C:\Miniconda3\envs\test\lib
C:\Miniconda3\envs\test
C:\Miniconda3\envs\test\lib\site-packages
```

I have access to multiple computers running conda and all show some version of this issue. I have also tried wiping miniconda and reinstalling to no avail.

## Expected Behavior
Obviously python should not be loading DLLs from other environments. It kind-of defeats the purpose. I've also found that if the environment it chooses to load from does not contain the necessary DLLs, import fails outright. Which environment it chooses to load from seems practically random at this point.

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```
     active environment : test
    active env location : C:\Miniconda3\envs\test
            shell level : 2
       user config file : C:\Users\Experiment Control\.condarc
 populated config files : 
          conda version : 4.6.14
    conda-build version : not installed
         python version : 3.7.3.final.0
       base environment : C:\Miniconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/win-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : C:\Miniconda3\pkgs
                          C:\Users\Experiment Control\.conda\pkgs
                          C:\Users\Experiment Control\AppData\Local\conda\conda\pkgs
       envs directories : C:\Miniconda3\envs
                          C:\Users\Experiment Control\.conda\envs
                          C:\Users\Experiment Control\AppData\Local\conda\conda\envs
               platform : win-64
             user-agent : conda/4.6.14 requests/2.21.0 CPython/3.7.3 Windows/7 Windows/6.1.7601
          administrator : False
             netrc file : None
           offline mode : False
```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```
# packages in environment at C:\Miniconda3\envs\test:
#
# Name                    Version                   Build  Channel
ca-certificates           2019.5.15                     0    defaults
certifi                   2019.3.9                 py37_0    defaults
openssl                   1.1.1c               he774522_1    defaults
pip                       19.1.1                   py37_0    defaults
psutil                    5.6.2            py37he774522_0    defaults
python                    3.7.3                h8c8aaf0_1    defaults
setuptools                41.0.1                   py37_0    defaults
sqlite                    3.28.0               he774522_0    defaults
vc                        14.1                 h0510ff6_4    defaults
vs2015_runtime            14.15.26706          h3a45250_4    defaults
wheel                     0.33.4                   py37_0    defaults
wincertstore              0.2                      py37_0    defaults
```
</p></details>

**Comment by msarahan** [comment]
It might be related up use of the old activate script somehow, though that would be very strange. The current way is to use ``conda activate``. This works after running conda init. I sort of wonder if you're picking up a different activate script, perhaps from virtualenv?

**Comment by dihm** [comment]
@msarahan 
I didn't know about the new activate command. 
I ran `conda init` and it noted a few changes. However, I get the same behavior using `conda activate` as I do with plain `activate`. That said, this certainly smells strongly of windows path stupidity.

My initial command prompt is being loaded with `%windir%\System32\cmd.exe "/K" C:\Miniconda3\Scripts\activate.bat C:\Miniconda3`

I believe my activate.bat is current: (ignoring comments)
```
@set "_args1=%1"
@set _args1_first=%_args1:~0,1%
@set _args1_last=%_args1:~-1%
@set _args1_first=%_args1_first:"=+%
@set _args1_last=%_args1_last:"=+%
@set _args1=

@if "%_args1_first%"=="+" if NOT "%_args1_last%"=="+" (
    @CALL "%~dp0..\condabin\conda.bat" activate
    @GOTO :End
)

@REM This may work if there are spaces in anything in %*
@CALL "%~dp0..\condabin\conda.bat" activate %*

:End
@set _args1_first=
@set _args1_last=
```

**Comment by dihm** [comment]
Have a bit more testing. It looks like I only see this behavior if there is more than one non-base environment with the same python version (I updated the Steps to Reproduce to reflect this).

Attempting this on multiple machines; which environments load incorrect DLLs is somewhat arbitrary. Behavior i've seen so far includes: only one DLL from the wrong environment, most DLLs from the wrong environment, and even one instance where behavior was correct in one environment but not the other [ie test works, test2 does not].

Every machine I've tested I've first run conda init to update the activation scripts. Tests include Windows 10 and 7 machines. I've been checking every path I can find with no luck so far.
